rsyslog/Sanity/gnutls-openssl/runtest.sh

```diff
@@ -0,0 +1,270 @@
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Include Beaker environment
+. /usr/bin/rhts-environment.sh || :
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+PACKAGE="rsyslog"
+
+rlJournalStart && {
+  rlPhaseStartSetup && {
+    rlRun "rlCheckMakefileRequires" || rlDie "cannot continue"
+    rlRun "rlImport --all" 0 "Import libraries" || rlDie "cannot continue"
+    CleanupRegister 'rlRun "rsyslogCleanup"'
+    rlRun "rsyslogSetup"
+    CleanupRegister 'rlRun "rsyslogServerCleanup"'
+    rlRun "rsyslogServerSetup"
+    rlRun "TmpDir=\$(mktemp -d)" 0 "Creating tmp directory"
+    CleanupRegister "rlRun 'rm -r $TmpDir' 0 'Removing tmp directory'"
+    CleanupRegister 'rlRun "popd"'
+    rlRun "pushd $TmpDir"
+
+
+    cat > ca.tmpl <<EOF
+organization = "Red Hat"
+unit = "GSS"
+locality = "Brno"
+state = "Moravia"
+country = CZ
+cn = "rsyslog+gnutls"
+serial = 001
+expiration_days = 365
+dns_name = "$(hostname)"
+ip_address = "127.0.0.1"
+email = "root@$(hostname)"
+crl_dist_points = "http://127.0.0.1/getcrl/"
+ca
+cert_signing_key
+crl_signing_key
+EOF
+    rlRun "certtool --generate-privkey --outfile ca-key.pem" 0 "Generate key for CA"
+    rlRun "certtool --generate-self-signed --load-privkey ca-key.pem --template ca.tmpl --outfile ca-cert.pem" 0 "Generate self-signed CA cert"
+
+    cat > server.tmpl <<EOF
+organization = "Red Hat"
+unit = "GSS"
+locality = "Brno"
+state = "Moravia"
+country = CZ
+cn = "rsyslog+gnutls"
+serial = 002
+expiration_days = 365
+dns_name = "$(hostname)"
+ip_address = "127.0.0.1"
+email = "root@$(hostname)
+tls_www_server
+EOF
+    cat server.tmpl
+    rlRun "certtool --generate-privkey --outfile server-key.pem --bits 2048" 0 "Generate key for server"
+    rlRun "certtool --generate-request --template server.tmpl --load-privkey server-key.pem --outfile server-request.pem" 0 "Generate server cert request"
+    rlRun "certtool --generate-certificate --template server.tmpl --load-request server-request.pem  --outfile server-cert.pem --load-ca-certificate ca-cert.pem --load-ca-privkey ca-key.pem" 0 "Generate server cert"
+
+    cat > client.tmpl <<EOF
+organization = "Red Hat"
+unit = "GSS"
+locality = "Brno"
+state = "Moravia"
+country = CZ
+cn = "rsyslog+gnutls"
+serial = 003
+expiration_days = 365
+dns_name = "$(hostname)"
+ip_address = "127.0.0.1"
+email = "root@$(hostname)"
+tls_www_server
```

Comment:
```suggestion
tls_www_server
tls_www_client
```

Reply:
this does not cause any troubles with EKU purpose disabled

Reply:
actually, there should not be both of them, I guess..

Reply:
It's possible to have only 1 of them but in that case it would be required to generate new certificates for each communication type(gtls->gtls, gtls->ossl,...). I think there is no harm if we let it there.
The implementation of rsyslog-gnutls package does not care about certificate purpose, however, rsyslog-openssl does. In case tls_www_client is not present in client side, certificate verification will fail.

Reply:
I think it is not necessary to regenerate it as the client is always the client and the server is always the server.

Reply:
Yes, I updated the suggested batch according to that.